### PR TITLE
DOC: Edit MultiIndex.set_levels() docstring to clarify that set_levels() interprets passed values as new components of the .levels attribute (#28294)

### LINKED ISSUE #28294: Clarify that MultiIndex.set_levels() interprets passed values as new components of the .levels attribute
#### Code Sample, a copy-pastable example if possible

```python
import numpy as np
import pandas as pd
df = pd.DataFrame(np.random.rand(3, 3), columns=pd.MultiIndex.from_tuples([(1, 2), (1, 4), (5, 6)]))
df.columns = df.columns.set_levels([1, 1, 3], level=0)
```
Which raises a ```ValueError```:

```python
ValueError: Level values must be unique: [1, 1, 3] on level 0
```

#### Problem description

Despite a dataframe with non-unique MultiIndex can be created, they cannot be set using ```set_levels()```. Is this behaviour expected? I believe non-unique level values were not allowed for a period of time (#18882), but then they were allowed again (#21423), so I am not sure which is the current convention.

#### Expected Output

```python
          1                   3
          2         4         6
0  0.317669  0.329142  0.056725
1  0.969472  0.340309  0.135204
2  0.242408  0.934748  0.683186
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.4.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 10, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 0.25.1
numpy            : 1.16.4
pytz             : 2019.2
dateutil         : 2.8.0
pip              : 19.2.2
setuptools       : 41.0.1
Cython           : 0.29.13
pytest           : 5.1.2
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 1.1.8
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.8.0
pandas_datareader: None
bs4              : None
bottleneck       : 1.2.1
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.1
numexpr          : None
odfpy            : None
openpyxl         : 2.6.2
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : 1.3.1
sqlalchemy       : 1.3.7
tables           : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : 1.1.8

</details>

### PR DESCRIPTION
**Pull request by hweecat**
Added documentation on MultiIndex.set_levels to clarify that MultiIndex.set_levels() interprets passed values as new components of the .levels attribute. Continuation of #28797 due to issues with Travis CI for Python 3.8 and subsequent git diff issues resulting from resolution of Travis CI issues.

- [x] closes #28294 - or at least attempts to
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by jbrockmendel** [comment]
can you rebase

**Comment by hweecat** [comment]
Hello @jbrockmendel, do you mean 'git rebase upstream/master' or 'git fetch upstream && git merge upstrea/master' to resolve conflicts?

**Comment by jbrockmendel** [comment]
> do you mean 'git rebase upstream/master' or 'git fetch upstream && git merge upstream/master' to resolve conflicts?

I usually do `git pull upstream master` and manually resolve any conflicts.

**Comment by hweecat** [comment]
Not sure what happened, but I did a `git fetch upstream`, 'git pull upstream/master`, manually resolved conflicts, and `git push origin` on my branch - and my tests are failing.

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -757,32 +769,66 @@ def set_levels(self, levels, level=None, inplace=False, verify_integrity=True):
         Examples
         --------
         >>> idx = pd.MultiIndex.from_tuples([(1, 'one'), (1, 'two'),
-                                            (2, 'one'), (2, 'two')],
+                                            (2, 'one'), (2, 'two'),
+                                            (3, 'one'), (3, 'two')],
                                             names=['foo', 'bar'])
-        >>> idx.set_levels([['a', 'b'], [1, 2]])
+        >>> idx
```
Comment (WillAyd) [inline pandas/core/indexes/multi.py]: I think you can remove this; very duplicative of the constructor directly preceding

File: pandas/core/indexes/multi.py [hunk 2]
```diff
@@ -757,32 +769,66 @@ def set_levels(self, levels, level=None, inplace=False, verify_integrity=True):
...
+                    (3, 'b')],
+                   names=['foo', 'bar'])
+
+        ``set_levels()`` passes values into the levels attribute that
+        is represented by a FrozenList containing list of values for
+        each level in the MultiIndex, even when the number of values
+        passed for a level is more than the number of indexes
+        available in the MultiIndex itself.
+
+        >>> idx.set_levels([['a', 'b', 'c'], [1, 2]], level=[0, 1])
```
Comment (WillAyd) [inline pandas/core/indexes/multi.py]: Can remove this as it is duplicative of what's above

File: pandas/core/indexes/multi.py [hunk 3]
```diff
@@ -757,32 +769,66 @@ def set_levels(self, levels, level=None, inplace=False, verify_integrity=True):
...
+        MultiIndex([('a', 1),
+                    ('a', 2),
+                    ('b', 1),
+                    ('b', 2),
+                    ('c', 1),
+                    ('c', 2)],
                    names=['foo', 'bar'])
-        >>> idx.set_levels([['a', 'b'], [1, 2]], level=[0, 1])
+        >>> idx.set_levels([['a', 'b', 'c'], [1, 2]], level=[0, 1]).levels
```
Comment (WillAyd) [inline pandas/core/indexes/multi.py]: Remove this too

File: pandas/core/indexes/multi.py [hunk 4]
```diff
@@ -757,32 +769,66 @@ def set_levels(self, levels, level=None, inplace=False, verify_integrity=True):
...
         MultiIndex([(1, 'a'),
                     (1, 'b'),
                     (2, 'a'),
-                    (2, 'b')],
+                    (2, 'b'),
+                    (3, 'a'),
+                    (3, 'b')],
+                   names=['foo', 'bar'])
+
+        ``set_levels()`` passes values into the levels attribute that
```
Comment (WillAyd) [inline pandas/core/indexes/multi.py]:
```suggestion
        If any of the levels passed to `set_levels` exceeds the existing length, all of the values from that argument will be stored though truncated in the MultiIndex output
```

File: pandas/core/indexes/multi.py [hunk 5]
```diff
@@ -739,6 +739,18 @@ def set_levels(self, levels, level=None, inplace=False, verify_integrity=True):
         """
         Set new levels on MultiIndex. Defaults to returning new index.
 
+        ``set_levels`` passes a new value for each index in the level,
```
Comment (WillAyd) [inline pandas/core/indexes/multi.py]: I think you can do away with the long description entirely

**Review by WillAyd — CHANGES_REQUESTED** [review]
Some minor edits; this is a rather tough concept to express so I think if we can make succinct will help explain better

File: pandas/core/indexes/multi.py (hunk 4 above)
Comment (hweecat) [reply pandas/core/indexes/multi.py]: Thanks @WillAyd for the feedback - I've been attempting to express the concept through examples and all, but your explanation is very succinct and neat! Will incorporate those changes you've suggested. :)

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
lgtm @gfyoung care to take a look?

File: pandas/core/indexes/multi.py [hunk 6]
```diff
@@ -757,32 +757,46 @@ def set_levels(self, levels, level=None, inplace=False, verify_integrity=True):
...
-                    (2, 'b')],
+                    (2, 'b'),
+                    (3, 'a'),
+                    (3, 'b')],
                    names=['foo', 'bar'])
-        >>> idx.set_levels([['a', 'b'], [1, 2]], level=[0, 1])
+
+        If any of the levels passed to ``set_levels()`` exceeds the
+        existing length, all of the values from that argument will
+        be stored though truncated in the MultiIndex output.
```
Comment (gfyoung) [inline pandas/core/indexes/multi.py]: This wording ("stored though truncated") is a little awkward

Comment (hweecat) [reply pandas/core/indexes/multi.py]: Would the following wording be clearer in expressing the concept?

"stored in the MultiIndex variable, though the values will be truncated in the MultiIndex output."

Feel free to let me know what you think.

Comment (WillAyd) [reply pandas/core/indexes/multi.py]: I think that’s a great edit - care to make that change?

**Comment by jreback** [comment]
can you merge master

**Comment by hweecat** [comment]
@jbrockmendel I've done a "git fetch upstream" and "git merge upstream/master" - let's hope tests pass this time.

**Comment by hweecat** [comment]
Okay, I have no idea what happened to my branch when I'm merging master, and tests are failing - suspect that it's caused by my previous commits made by merging master.

Anyone has any idea what to do with this?

**Comment by WillAyd** [comment]
Did you merge `upstream/master` or master from your branch? I'm guessing the latter given the amount of files changed

You can maybe reset hard back to before the merge and try again with `upstream/master`

**Comment by hweecat** [comment]
Just checked - tests finally passed after the reset hard and merge with `upstream master`. Thanks for the help!

**Comment by datapythonista** [comment]
Thanks for the work on this @hweecat

**Pull request merged**